From 9783991b615484d2926e9648b10ea090af81d93f Mon Sep 17 00:00:00 2001
From: Mitch Curtis <mitch.curtis@theqtcompany.com>
Date: Wed, 3 Feb 2016 12:57:05 +0100
Subject: [PATCH 08/61] Fix crash when Canvas has negative width or height

m_fbo is null when using a threaded render loop.

Change-Id: I297ba651f9605f1718dbe9d09bd30e9682fb8401
Task-number: QTBUG-50085
Reviewed-by: Laszlo Agocs <laszlo.agocs@theqtcompany.com>
---
 src/quick/items/context2d/qquickcontext2dtexture.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/quick/items/context2d/qquickcontext2dtexture.cpp b/src/quick/items/context2d/qquickcontext2dtexture.cpp
index 73b0b55..5ac971d 100644
--- a/src/quick/items/context2d/qquickcontext2dtexture.cpp
+++ b/src/quick/items/context2d/qquickcontext2dtexture.cpp
@@ -599,6 +599,11 @@ QPaintDevice* QQuickContext2DFBOTexture::beginPainting()
 void QQuickContext2DFBOTexture::endPainting()
 {
     QQuickContext2DTexture::endPainting();
+
+    // There may not be an FBO due to zero width or height.
+    if (!m_fbo)
+        return;
+
     if (m_multisampledFbo)
         QOpenGLFramebufferObject::blitFramebuffer(m_fbo, m_multisampledFbo);
 
-- 
1.9.3

